src/model/window-guardian.ts

```diff
@@ -68,25 +68,21 @@ const makeWindowGuardianConfig = (
 
 export const makeGuardianBrowserCAPI = (CAPI: CAPIType): CAPIBrowserType => {
 	// For some elements it is important to keep thier index in the `elements` array
-
 	const blockElementWithIndex = <T extends CAPIElement>(
 		blocks: { elements: CAPIElement[] }[],
 		blockElementType: CAPIElement['_type'],
 		indexName: string,
 	): T[] => {
 		if (blocks.length === 0) return [];
-		return blocks[0].elements.reduce(
-			(acc: T[], element: CAPIElement, index: number) => {
-				if (element._type === blockElementType) {
-					acc.push({
-						...element,
-						[indexName]: index,
-					} as T);
-				}
-				return acc;
-			},
-			[] as T[],
```

Comment:
I feel like this is the key bit. If your accumulator is the same type as your original data structure, I wonder what the goal of using `reduce` instead of `map` is

Reply:
The goal is that this actually need `filter().map()`